lucee-java/lucee-core/src/lucee/commons/io/res/type/s3/S3Resource.java

```diff
@@ -69,6 +70,44 @@ public final class S3Resource extends ResourceSupport {
 		this.objectName = objectName.toLowerCase(Locale.ENGLISH);
 	}
 
+	public void putFile(FileResource fileResource) {
+		s3.put(objectName, fileResource);
+	}
+
+	public void downloadFile(FileResource fileResource) {
+		s3.downloadFile(objectName, fileResource);
+	}
+
+	@Override
+	public void copyFrom(Resource res,boolean append) throws IOException {
+		if (res instanceof S3Resource && !append) {
```

Comment:
can we put these if conditionals as their own method in a utility class? `isS3Resource` etc.